workflow/executor/executor.go

```diff
@@ -217,17 +214,17 @@ func (we *WorkflowExecutor) SaveParameters() error {
 	}
 
 	for i, param := range we.Template.Outputs.Parameters {
-		log.Infof("Saving out parameter: %s", param.Name)
+		log.Infof("Saving path output parameter: %s", param.Name)
 		// Determine the file path of where to find the parameter
-		if param.Path == "" {
-			return errors.InternalErrorf("Output parameter %s did not specify a file path", param.Name)
+		if param.ValueFrom == nil || param.ValueFrom.Path == "" {
+			continue
 		}
 		// Use docker cp command to print out the content of the file
 		// Node docker cp CONTAINER:SRC_PATH DEST_PATH|- streams the contents of the resource
 		// as a tar archive to STDOUT if using - as DEST_PATH. Thus, we need to extract the
 		// content from the tar archive and output into stdout. In this way, we do not need to
 		// create and copy the content into a file from the wait container.
-		dockerCpCmd := fmt.Sprintf("docker cp -a %s:%s - | tar -ax -O", mainCtrID, param.Path)
+		dockerCpCmd := fmt.Sprintf("docker cp -a %s:%s - | tar -ax -O", mainCtrID, param.ValueFrom.Path)
```

Comment:
Should we use `kubectl cp` instead to keep this container runtime agnostic?

Reply:
kubectl requires api service access instead of host docker access, not to mention we would then need operate on a staged temporary file instead of the stdout stream. It's an unnecessary complication for a problem that has no practical need, given the current state of kubernetes. Even if we cared about supporting other runtimes, I imagine we would want to do the equivalent of what we are doing using their tooling.